### PR TITLE
doc: How to configure Airbrake gem (in the Readme and code)

### PR DESCRIPTION
**Pull request by munen**
As discussed in https://github.com/errbit/errbit/pull/1514, harmonize the Readme and generated documentation.

### CODE REVIEW

File: app/views/apps/_configuration_instructions.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  config.project_key = '<%= api_key %>'
+
+  # Airbrake supports polling a remote config from airbrake.io,
+  # enabled by default. However, since you are using Errbit and not
+  # Airbrake, this request would fail and generate an XML error
+  # message on stdout.
+  config.remote_config = false
+
+  # The same goes for performance_stats
+  config.performance_stats = false
```
Comment (burnettk) [inline app/views/apps/_configuration_instructions.html.erb:22]: what about these two, or are they not needed? https://gist.github.com/2called-chaos/07a62137b75a4bf2261a328bc94ed26f#file-z_airbrake-rb-L32-L33

Comment (munen) [reply app/views/apps/_configuration_instructions.html.erb]: They are certainly fine. I've realized it at the same time you wrote this. Rebased and pushed.

**Comment by munen** [comment]
@2called-chaos provided a [nice gist](https://gist.github.com/2called-chaos/07a62137b75a4bf2261a328bc94ed26f#file-z_airbrake-rb-L32-L33) with an elaborate config.

I omitted most of the flags, because:

1. I wanted to keep the docs to a minimum.
2. Since we document using the generator, the users will always have up to date flags.
3. Testing the more complicated (and probably as beneficial) flags is outside the scope of what I can do right now. It's probably also better if a core maintainer can take a look at them.

**Comment by burnettk** [comment]
there's a lot of good stuff in the gist in terms of making useful stuff from the request env available, filtering out less useful stuff from the request env, ignoring less useful exceptions, etc. i went through and grabbed some stuff for our env, but i'm not sure it's the sort of thing we can document as "you will definitely want this in your configs" unless we make it a lot smaller. i'd definitely be open to a pared-down version of this that we could add to the new configuration instructions that just includes the uncontroversial bits, in case someone wants to take a stab at that.

**Comment by 2called-chaos** [comment]
It's been a long time since I've used airbrake v5 but I remember being rather annoyed with v6 as it reported literally everything and without any useful annotation data like v5 had. To be frank I haven't tried an "empty" config recently, does it still omit env and headers by default? I would value this as pretty important and maybe that is worth including then.

Ignoring exceptions can be important, it certainly is for us as we would get thousands of AR::NotFound alone for example. I suppose this isn't as big of a deal if the backend supports ignoring errors which Errbit does not (#1507) but depending on how such an ignore function would work it might still get spammy as hell if an unnice person deems it necessary to vulnerability scan your site (which raises many errors in Rails' middlewares so you can't even rescue in your app).

But there is a reason I just posted my config from time to time rather than make a doc out of it, I don't know what normal people want or need (I tend to go overboard with options and flexibility). And certainly do I not have a good idea of where to place it. I too want to keep the instructions and readme as concise as possible but maybe we can link to/hint at a doc file in the repo or wiki with "advanced config examples" or something along those lines.

But then again most if not all [is already kind of documented](https://github.com/airbrake/airbrake-ruby#config-options) but from what I gathered from you guys is that nobody really wants to read all this (myself included). While skimming through I just learned that setting `performance_stats` to false makes it unnecessary to disable the other stats separately.

So all in all maybe we just link to the official documentation and maybe provide a (cleaned up) version of my gist as inspiration for what to ctrl-F for :D

**Comment by burnettk** [comment]
i think i'm on the same page with you on pretty much all of this. https://github.com/airbrake/airbrake/blob/master/CHANGELOG.md#v610rc1-may-10-2017 talks about the "omit env and headers by default" thing. i've always wondered where those went, thanks for the lead about it changing in airbrake 6. so we should probably re-shove some stuff from the context back into the env in our example configs somehow.

**Pull request merged**